Dockerfile

```diff
@@ -38,3 +46,9 @@ COPY --from=builder /workspace/manager .
 RUN apk update && apk add ca-certificates bash curl drill jq
 
 ENTRYPOINT ["/manager"]
+
+FROM gcr.io/distroless/static:nonroot as backup-agent
+WORKDIR /
+COPY --from=builder /workspace/backup-agent .
+USER nonroot:nonroot
+ENTRYPOINT ["/backup-agent"]
```

Comment:
For the implementation of the proxy (https://github.com/improbable-eng/etcd-cluster-operator/tree/master/build/package) and the plan for the restore agent (https://github.com/improbable-eng/etcd-cluster-operator/issues/146) I'd put separate Dockerfiles under the path `build/package` in-line with the standard [Go project layout](https://github.com/golang-standards/project-layout#build).

Doing this as multiple targets in one Dockerfile isn't my preference (by making the proxy Dockerfile separate I made it much simpler, I suspect we'd end up with a very complex single file if we combined all of the output builds in it). But it's not "wrong" either.

I'd like us to be consistent. So either:

* Pull the proxy's build into the main Dockerfile, delete `build/package`, and update #146.
* Pull the backup agent's build into it's own Dockerfile under `build/package/backup-agent.Dockerfile`

Reply:
Yeah, I know. This is  a WIP branch, not yet ready for review.

I wanted to try this out.
The advantage is that we only have to create the build environment once and define the `ldflags=-X=...` and other build configuration once rather than duplicating it in every Dockerfile.
Also that all the binaries are compiled from exactly the same objects.

But I agree that having separate dockerfiles is clearer.

Reply:
Done.